### PR TITLE
docs(linux): Update packaging documentation

### PR DESCRIPTION
**Pull request by ermshiperete**
- remove obsolete `packageName` variable
- add section on how to build with Docker, similar to how the package build is done on GHA
- updated "uploading to Debian" section
- some reformatting

@keymanapp-test-bot skip

### CODE REVIEW

File: docs/linux/packaging.md [hunk 1]
```diff
@@ -169,38 +169,81 @@ Building packages happen in the [Keyman source tree](https://github.com/keymanap
...
-This creates a source package (`<packageName>_<version>-1.dsc`) and some `*.tar.?z` files in the source root directory for `keyman`.
+This creates a source package (`keyman_<version>-1.dsc`) and some `*.tar.?z`
+files in the source root directory for `keyman`.
 
 ci-builder-script's [`build-package`](https://github.com/sillsdev/ci-builder-scripts/blob/master/bash/build-package)
 script creates the binary packages:
 
 ```bash
-cd linux/${packageName}
+cd $KEYMAN_ROOT
```
Comment (darcywong00) [inline docs/linux/packaging.md:186]: So we don't cd to `$KEYMAN_ROOT/linux` ?

Comment (ermshiperete) [reply docs/linux/packaging.md:186]: No, because the source package gets put in `$KEYMAN_ROOT`, so we need to be in that directory.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.48-alpha](https://keyman.com/downloads/releases/alpha/17.0.48)

**Pull request merged**